examples/custom-usage-counter/custom_usage_counter.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+import argparse
+import asyncio
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from decimal import Decimal
+from pathlib import Path
+
+from yapapi.strategy import LeastExpensiveLinearPayuMS, DecreaseScoreForUnconfirmedAgreement
+
+from yapapi.log import enable_default_logger
+
+from yapapi.props.base import prop, constraint
+from yapapi.props import inf, com
+
+from yapapi import Golem
+from yapapi.ctx import ActivityUsage
+from yapapi.payload import Payload
+from yapapi.services import Service, ServiceState
+
+
+@dataclass
+class CustomCounterServicePayload(Payload):
+    runtime: str = constraint(inf.INF_RUNTIME_NAME, default="test-counters")
+    min_mem_gib: float = constraint(inf.INF_MEM, operator=">=", default=0.5)
+    min_storage_gib: float = constraint(inf.INF_STORAGE, operator=">=", default=0.1)
+
+
+class CustomCounterService(Service):
+    @staticmethod
+    async def get_payload():
+        return CustomCounterServicePayload()
+
+    def __init__(self, *args, running_time_sec, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._running_time_sec = running_time_sec
+
+    async def start(self):
+        self._ctx.run("sleep", "1")
+        yield self._ctx.commit()
+
+    async def run(self):
+        start_time = datetime.now()
+        print(f"service {self.id} running on '{self.provider_name}'...")
+        while datetime.now() < start_time + timedelta(seconds=self._running_time_sec):
+            self._ctx.run("sleep", "1000")
```

Comment:
why do we need to issue this `sleep` command all the time? does it influence the custom conuter?

Reply:
I think it's a good idea to issue this command, because the purpose of this example is to show that we can run commands and then get the custom counter value from the provider.

https://github.com/golemfactory/ya-test-runtime-counters/blob/main/src/main.rs

This particular runtime increases the counter every two seconds even if no commands are issued.